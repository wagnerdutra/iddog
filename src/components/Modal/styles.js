import styled from 'styled-components';

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.8);
  z-index: 1;
  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: center;
  top: 0;
  left: 0;
`;

export { Container };
