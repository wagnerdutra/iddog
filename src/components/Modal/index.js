import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import { Container } from './styles';

import CSSTransitionEF from '../CSSTransitionEF';

function Modal({ closeModal, children, showModal }) {
  const containerRef = useRef();

  function handleContainerClick(e) {
    if (e.target !== containerRef.current) return;
    closeModal();
  }

  return (
    <CSSTransitionEF inProp={showModal}>
      <Container
        ref={containerRef}
        onClick={handleContainerClick}
        showModal={showModal}
        data-testid="modal-id"
      >
        {children}
      </Container>
    </CSSTransitionEF>
  );
}

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  closeModal: PropTypes.func.isRequired,
  showModal: PropTypes.bool.isRequired
};

export default Modal;
