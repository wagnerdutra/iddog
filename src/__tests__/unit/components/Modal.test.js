import React from 'react';

import { render, fireEvent } from 'react-testing-library';

import Modal from 'Components/Modal';

describe('<Modal />', () => {
  it('should render with show false correctly', () => {
    const { container } = render(
      <Modal closeModal={() => {}} showModal={false}>
        <p>Teste</p>
      </Modal>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with show true correctly', () => {
    const { container } = render(
      <Modal closeModal={() => {}} showModal>
        <p>Teste</p>
      </Modal>
    );
    expect(container).toMatchSnapshot();
  });

  it('should handle modal click', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <Modal closeModal={onClick} showModal>
        <p>Teste</p>
      </Modal>
    );
    fireEvent.click(getByTestId('modal-id'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
