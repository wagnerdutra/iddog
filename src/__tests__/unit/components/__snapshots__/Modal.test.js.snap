// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render with show false correctly 1`] = `<div />`;

exports[`<Modal /> should render with show true correctly 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: rgba(0,0,0,0.8);
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  top: 0;
  left: 0;
}

<div>
  <div
    class="c0"
    data-testid="modal-id"
  >
    <p>
      Teste
    </p>
  </div>
</div>
`;
